src/agentlab2/base.py

```diff
@@ -0,0 +1,34 @@
+"""Base classes for AgentLab2."""
+
+import importlib
+
+from pydantic import BaseModel, model_serializer, model_validator
+
+
+class AL2BaseModel(BaseModel):
+    """
+    Base class for Pydantic models that can save and load their type information.
+
+    When serialized, includes `_type` field with the fully qualified class name.
+    When deserialized, uses `_type` to instantiate the correct subclass.
```

Comment:
Does this achieve the same effect?
https://docs.pydantic.dev/2.3/usage/types/string_types/#importstring